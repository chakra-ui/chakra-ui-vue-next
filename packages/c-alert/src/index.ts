import { defineComponent, computed, createVNode, mergeProps, isVNode, Fragment } from 'vue';
import { createStylesContext, useMultiStyleConfig, chakra } from '@chakra-ui/vue-system';
import { createContext, getValidChildren } from '@chakra-ui/vue-utils';
import { createIconComponent } from '@chakra-ui/c-icon';

function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function (target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];

      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }

    return target;
  };
  return _extends.apply(this, arguments);
}

var CInfoIcon = createIconComponent("info");
var CCheckIcon = createIconComponent("check-circle");
var CWarningIcon = createIconComponent("warning-alt");
var CErrorIcon = createIconComponent("warning");

function _isSlot(s) {
  return typeof s === 'function' || Object.prototype.toString.call(s) === '[object Object]' && !isVNode(s);
}

var STATUSES = {
  info: {
    colorScheme: "blue",
    icon: CInfoIcon
  },
  success: {
    colorScheme: "green",
    icon: CCheckIcon
  },
  warning: {
    colorScheme: "orange",
    icon: CWarningIcon
  },
  error: {
    colorScheme: "red",
    icon: CErrorIcon
  },
  loading: {
    icon: CInfoIcon,
    colorScheme: "blue"
  }
};

var _createStylesContext = createStylesContext("Alert"),
    StylesProvider = _createStylesContext[0],
    useStyles = _createStylesContext[1];

var _createContext = createContext({
  name: "AlertContext",
  errorMessage: "useAlertContext: `context` is undefined. Seems you forgot to wrap alert components in `<c-alert />`"
}),
    AlertProvider = _createContext[0],
    useAlertContext = _createContext[1];
/**
 * CAlert component
 *
 * This is the container component for all Alert components.
 * It also provides state and context to it's compound components
 */


var CAlert = defineComponent({
  name: "CAlert",
  props: {
    as: {
      type: [String, Object],
      "default": "div"
    },
    status: {
      type: [String],
      "default": "info"
    },
    colorScheme: {
      type: [String]
    },
    styleConfig: {
      type: [Object]
    },
    variant: {
      type: [String],
      "default": "solid"
    }
  },
  setup: function setup(props, _ref) {
    var slots = _ref.slots,
        attrs = _ref.attrs;
    var colorScheme = computed(function () {
      var _STATUSES$props$statu;

      return props.colorScheme || ((_STATUSES$props$statu = STATUSES[props == null ? void 0 : props.status]) == null ? void 0 : _STATUSES$props$statu.colorScheme);
    });
    var themingProps = computed(function () {
      return {
        colorScheme: colorScheme.value,
        variant: props.variant
      };
    });
    AlertProvider({
      status: computed(function () {
        return props.status;
      })
    });
    var styles = useMultiStyleConfig("Alert", themingProps);
    StylesProvider(styles);
    var alertStyles = computed(function () {
      return _extends({
        width: "100%",
        display: "flex",
        alignItems: "center",
        position: "relative",
        overflow: "hidden"
      }, styles.value.container);
    });
    return function () {
      return createVNode(chakra.div, mergeProps({
        "role": "alert",
        "__label": "alert",
        "__css": alertStyles.value
      }, attrs), {
        "default": function _default() {
          return getValidChildren(slots);
        }
      });
    };
  }
});
/**
 * CAlertTitle component
 *
 * The title component for alerts
 */

var CAlertTitle = defineComponent({
  name: "CAlertTitle",
  setup: function setup(_, _ref2) {
    var attrs = _ref2.attrs,
        slots = _ref2.slots;
    var styles = useStyles();
    return function () {
      return createVNode(chakra.div, mergeProps({
        "__label": "alert__title",
        "__css": styles.value.title
      }, attrs), _isSlot(slots) ? slots : {
        "default": function _default() {
          return [slots];
        }
      });
    };
  }
});
/**Â«
 * CAlertDescription component
 *
 * The description component for alerts
 */

var CAlertDescription = defineComponent({
  name: "CAlertDescription",
  setup: function setup(_, _ref3) {
    var attrs = _ref3.attrs,
        slots = _ref3.slots;
    var styles = useStyles();
    return function () {
      return createVNode(chakra.div, mergeProps({
        "__label": "alert__description",
        "__css": styles.value.description
      }, attrs), {
        "default": function _default() {
          return getValidChildren(slots);
        }
      });
    };
  }
});
/**
 * CAlertIcon component
 *
 * The Icon component for alerts
 */

var CAlertIcon = defineComponent({
  name: "CAlertIcon",
  setup: function setup(_, _ref4) {
    var attrs = _ref4.attrs,
        slots = _ref4.slots;
    var styles = useStyles();

    var _useAlertContext = useAlertContext(),
        status = _useAlertContext.status;

    var BaseIcon = STATUSES[status.value].icon;
    var css = computed(function () {
      return status.value === "loading" ? styles.value.spinner : styles.value.icon;
    });
    return function () {
      var validChildren = getValidChildren(slots);
      return createVNode(chakra.span, mergeProps({
        "display": "inherit",
        "__label": "alert__icon"
      }, attrs, {
        "__css": css.value
      }), {
        "default": function _default() {
          return createVNode(Fragment, null, [validChildren.length ? slots : createVNode(BaseIcon, {
            "h": "100%",
            "w": "100%"
          }, null)]);
        }
      }); // return <icon {...styles.value.icon} {...attrs}></icon>
    };
  }
});

export { CAlert, CAlertDescription, CAlertIcon, CAlertTitle };
