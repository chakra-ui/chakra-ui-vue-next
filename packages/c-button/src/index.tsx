import { defineComponent } from "vue"

export { default as CButton } from "./button"
export { default as CButtonGroup } from "./button-group"
export { default as CIconButton } from "./icon-button"
