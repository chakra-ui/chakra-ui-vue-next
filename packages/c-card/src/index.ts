export { CCard, type CCardProps } from "./c-card"
export { CCardBody, type CCardBodyProps } from "./c-card-body"
export { CCardHeader, type CCardHeaderProps } from "./c-card-header"
