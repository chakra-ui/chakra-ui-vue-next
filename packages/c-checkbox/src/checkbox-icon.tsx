import { MotionDirective, useMotions } from "@vueuse/motion"
import {
  defineComponent,
  h,
  computed,
  onBeforeUnmount,
  PropType,
  withDirectives,
  watch,
} from "vue"
import { chakra } from "@chakra-ui/vue-system"
import { CAnimatePresence, TransitionDefaults } from "@chakra-ui/c-motion"
import { CMotion } from "@chakra-ui/c-motion"
import { genId } from "@chakra-ui/vue-utils"

export const CheckIcon = defineComponent({
  name: "CheckboxCheckIcon",
  setup(_, { attrs, slots }) {
    const transitionId = `check-icon-${genId()}`
    /** Handles exit transition */
    const leave = (done: VoidFunction) => {
      const motions = useMotions()
      const instance = motions[transitionId]
      instance?.leave(() => {
        done()
      })
    }
    onBeforeUnmount(() => {
      leave(() => null)
    })
    return () =>
      withDirectives(
        <chakra.svg
          width="1.2em"
          viewBox="0 0 12 10"
          style={{
            fill: "none",
            strokeWidth: 2,
            stroke: "currentColor",
            strokeDasharray: 16,
          }}
          {...attrs}
        >
          {() => <polyline points="1.5 6 4.5 9 10.5 1" />}
        </chakra.svg>,
        [
          [
            MotionDirective({
              initial: {
                opacity: 0,
                strokeDashoffset: 16,
                transition: TransitionDefaults.leave,
              },
              enter: {
                opacity: 1,
                strokeDashoffset: 0,
                transition: TransitionDefaults.enter,
              },
              leave: {
                opacity: 0,
                strokeDashoffset: 16,
                transition: TransitionDefaults.leave,
              },
            }),
            transitionId,
          ],
        ]
      )
  },
})

export const IndeterminateIcon = defineComponent({
  name: "CheckboxIndeterminateIcon",
  setup(_, { attrs, slots }) {
    const transitionId = `indeterminate-icon-${genId()}`
    /** Handles exit transition */
    const leave = (done: VoidFunction) => {
      const motions = useMotions()
      const instance = motions[transitionId]
      instance?.leave(() => {
        done()
      })
    }
    onBeforeUnmount(() => {
      leave(() => null)
    })
    return () =>
      withDirectives(
        <chakra.svg
          width="1.2em"
          viewBox="0 0 24 24"
          style={{ stroke: "currentColor", strokeWidth: 4 }}
          {...attrs}
        >
          {() => <line x1="21" x2="3" y1="12" y2="12" />}
        </chakra.svg>,
        [
          [
            MotionDirective({
              initial: {
                scaleX: 0.65,
                opacity: 0,
              },
              enter: {
                scaleX: 1,
                opacity: 1,
                transition: {
                  scaleX: { duration: 150 },
                  opacity: { duration: 150 },
                  ...TransitionDefaults.enter,
                },
              },
              leave: {
                scaleX: 0.65,
                opacity: 0,
                transition: TransitionDefaults.leave,
              },
            }),
            transitionId,
          ],
        ]
      )
  },
})

export const CCheckboxTransition = defineComponent({
  name: "CCheckboxTransition",
  props: {
    open: Boolean as PropType<boolean>,
  },
  setup(props, { attrs, slots }) {
    return () => (
      <CMotion
        type={"scale"}
        style={{
          display: "flex",
          alignItems: "center",
          justifyContent: "center",
          height: "100%",
        }}
      >
        {props.open ? () => <chakra.div>{slots}</chakra.div> : null}
      </CMotion>
    )
  },
})

export interface CheckboxIconProps {
  isIndeterminate?: boolean
  isChecked?: boolean
}

export const CheckboxIcon = defineComponent(
  (props: CheckboxIconProps, { slots, attrs }) => {
    const IconEl = computed(() =>
      props.isIndeterminate ? IndeterminateIcon : CheckIcon
    )

    return () => (
      <CCheckboxTransition open={props.isChecked || props.isIndeterminate}>
        <IconEl.value {...attrs} />
      </CCheckboxTransition>
    )
  }
)

CheckboxIcon.name = "CheckboxIcon"
CheckboxIcon.props = {
  isIndeterminate: Boolean as PropType<boolean>,
  isChecked: Boolean as PropType<boolean>,
}
