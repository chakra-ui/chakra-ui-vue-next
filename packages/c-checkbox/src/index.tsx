export { CCheckbox } from "./checkbox"
export {
  CCheckboxGroup,
  useCheckboxGroupContext,
  type CheckboxGroupContext,
} from "./checkbox-group"

export type { CCheckboxControlProps, CCheckboxProps } from "./checkbox"
export type { CCheckboxGroupProps } from "./checkbox-group"
