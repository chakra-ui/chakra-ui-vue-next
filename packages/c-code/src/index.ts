export { default as CCode } from './code'
