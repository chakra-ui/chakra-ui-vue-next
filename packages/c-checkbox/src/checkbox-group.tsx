import { UseCheckboxGroupProps } from "./checkbox.types"
import { ThemingProps } from "@chakra-ui/styled-system"
import {
  defineComponent,
  h,
  ComputedRef,
  PropType,
  computed,
  renderSlot,
} from "vue"
import { createContext, vueThemingProps } from "@chakra-ui/vue-utils"
import { ComponentWithProps, DeepPartial } from "@chakra-ui/vue-system"
import { useCheckboxGroup } from "./use-checkbox-group"

export interface CCheckboxGroupProps
  extends UseCheckboxGroupProps,
    Omit<ThemingProps<"Checkbox">, "orientation"> {}

export type CheckboxGroupContext = ComputedRef<
  ThemingProps & {
    isDisabled?: boolean
    value: (string | number)[]
    handleChange(value: number | string, isChecked: boolean): void
  }
>

const [CheckboxGroupProvider, useCheckboxGroupContext] =
  createContext<CheckboxGroupContext>({
    strict: false,
    name: "CheckboxGroupContext",
  })

export const CCheckboxGroup: ComponentWithProps<
  DeepPartial<CCheckboxGroupProps>
> = defineComponent({
  name: "CCheckboxGroup",
  props: {
    modelValue: {
      type: Object as PropType<(string | number)[]>,
      // eslint-disable-next-line vue/require-valid-default-prop
      default: () => [],
    },
    isDisabled: {
      type: Boolean as PropType<boolean>,
      default: false,
    },
    ...vueThemingProps,
  },
  emits: ["change", "update:modelValue"],
  setup(props, { emit, slots }) {
    const checkBoxGroupValue = computed<(number | string)[]>({
      get() {
        return props.modelValue
      },
      set(value) {
        emit("update:modelValue", value)
      },
    })

    const { modelValue, handleChange } = useCheckboxGroup({
      modelValue: checkBoxGroupValue,
      isDisabled: computed(() => props.isDisabled),
    })

    const checkboxGroupContext = computed(() => ({
      size: props.size,
      variant: props.variant,
      isDisabled: props.isDisabled,
      colorScheme: props.colorScheme,
      value: modelValue.value,
      handleChange,
    }))

    CheckboxGroupProvider(checkboxGroupContext)
    return () => renderSlot(slots, "default")
  },
})

export { useCheckboxGroupContext }
