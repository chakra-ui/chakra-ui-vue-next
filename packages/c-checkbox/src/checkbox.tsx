/**
 * Hey! Welcome to @chakra-ui/vue-next Checkbox
 *
 * C checkbox component is used in forms when a user needs to select multiple values from several options
 *
 * @see Docs     https://next.vue.chakra-ui.com/c-checkbox
 * @see Source   https://github.com/chakra-ui/chakra-ui-vue-next/blob/main/packages/c-checkbox/src/c-checkbox/c-checkbox.tsx
 * @see WAI-ARIA https://www.w3.org/TR/wai-aria-practices-1.2
 */

import {
  h,
  defineComponent,
  PropType,
  computed,
  onMounted,
  Fragment,
  cloneVNode,
  watch,
} from "vue"
import { chakra, useMultiStyleConfig } from "@chakra-ui/vue-system"
import {
  omitThemingProps,
  SystemStyleObject,
  ThemingProps,
  SystemProps,
} from "@chakra-ui/styled-system"
import {
  SNAO,
  vueThemingProps,
  getValidChildren,
  genId,
} from "@chakra-ui/vue-utils"
import { HTMLChakraProps } from "@chakra-ui/vue-system"
import * as checkbox from "@zag-js/checkbox"
import { normalizeProps, useMachine, mergeProps } from "@zag-js/vue"
import { useId, useIds } from "@chakra-ui/vue-composables"
import { filterUndefined, mergeWith, pick } from "@chakra-ui/utils"
import {
  CFormControlProviderContext,
  useFormControlContext,
} from "@chakra-ui/c-form-control"
import { CheckboxIcon } from "./checkbox-icon"
import { CheckboxGroupContext, useCheckboxGroupContext } from "./checkbox-group"

/**
 * - Implement checkbox as state machine.
 * - Implement checkbox as part of group.
 */

const CCheckboxControl = chakra("span", {
  baseStyle: {
    display: "inline-flex",
    alignItems: "center",
    justifyContent: "center",
    verticalAlign: "top",
    userSelect: "none",
    flexShrink: 0,
  },
})

const CLabel = chakra("label", {
  baseStyle: {
    cursor: "pointer",
    display: "inline-flex",
    alignItems: "center",
    verticalAlign: "top",
    position: "relative",
  },
})

export interface CCheckboxControlProps {
  isIndeterminate?: boolean
  /**
   * If `true`, the checkbox will be disabled
   */
  isDisabled?: boolean
  /**
   * If `true` and `disabled` is passed, the checkbox will
   * remain tabbable but not interactive
   */
  isFocusable?: boolean
  /**
   * If `true`, the checkbox will be readonly
   */
  isReadonly?: boolean
  /**
   * If `true`, the checkbox is marked as invalid.
   */
  isInvalid?: boolean
  /**
   * If `true`, the checkbox input is marked as required,
   */
  isRequired?: boolean
  /**
   * The name of the input field in a checkbox
   * (Useful for form submission).
   */
  name?: string
  /**
   * The value to be used in the checkbox input.
   * This is the value that will be returned on form submission.
   */
  value?: string | number
  /**
   * Defines the string that labels the checkbox element.
   */
  "aria-label"?: string
  "aria-labelledby"?: string
  "aria-invalid"?: boolean
  "aria-describedby"?: string
  tabIndex?: number | string
}

export interface CCheckboxProps
  extends HTMLChakraProps<"input">,
    ThemingProps<"Checkbox">,
    CCheckboxControlProps {
  /**
   * The spacing between the checkbox and its label text
   * @default 0.5rem
   * @type SystemProps["marginLeft"]
   */
  spacing?: SystemProps["marginLeft"]
  /**
   * The color of the checkbox icon when checked or indeterminate
   */
  iconColor?: string
  /**
   * The size of the checkbox icon when checked or indeterminate
   */
  iconSize?: string | number
  /**
   * The checked icon to use
   */
  icon?: any
  /**
   * Additional props to be forwarded to the `input` element
   */
  inputProps?: HTMLChakraProps<"input">
}

export const CCheckbox = defineComponent({
  name: "CCheckbox",
  props: {
    modelValue: {
      type: Boolean as PropType<boolean>,
      default: false,
    },
    value: [String, Number] as PropType<CCheckboxProps["value"]>,
    id: String as PropType<string>,
    isIndeterminate: Boolean as PropType<CCheckboxProps["isIndeterminate"]>,
    isFocusable: Boolean as PropType<CCheckboxProps["isFocusable"]>,
    isRequired: Boolean as PropType<CCheckboxProps["isRequired"]>,
    isInvalid: Boolean as PropType<CCheckboxProps["isInvalid"]>,
    isDisabled: Boolean as PropType<CCheckboxProps["isDisabled"]>,
    isReadonly: Boolean as PropType<CCheckboxProps["isReadonly"]>,
    defaultChecked: Boolean as PropType<boolean>,
    name: String as PropType<CCheckboxProps["name"]>,
    "aria-label": String as PropType<CCheckboxProps["aria-label"]>,
    "aria-labelledby": String as PropType<CCheckboxProps["aria-labelledby"]>,
    "aria-invalid": Boolean as PropType<CCheckboxProps["aria-invalid"]>,
    "aria-describedby": String as PropType<CCheckboxProps["aria-describedby"]>,
    spacing: {
      type: SNAO as PropType<CCheckboxProps["spacing"]>,
      default: "0.5rem",
    },
    iconColor: String as PropType<CCheckboxProps["iconColor"]>,
    iconSize: [String, Number] as PropType<CCheckboxProps["iconSize"]>,
    icon: Object as PropType<CCheckboxProps["icon"]>,
    inputProps: Object as PropType<CCheckboxProps["inputProps"]>,
    ...vueThemingProps,
  },
  emits: ["change", "update:modelValue"],
  setup(props, { slots, attrs, emit }) {
    const group = useCheckboxGroupContext(
      computed(() => ({})) as CheckboxGroupContext
    )
    const ownProps = computed(() => omitThemingProps(props))
    const mergedProps = computed(() => mergeWith({}, group.value, props, attrs))
    const styles = useMultiStyleConfig("Checkbox", mergedProps)

    const inheritedFormControlProps = useFormControlContext(
      mergedProps as CFormControlProviderContext
    )

    const id = genId()
    const [rootId, inputId, controlId, labelId] = useIds(
      `chakra-checkbox-${id}`,
      "root",
      "input",
      "control",
      "label"
    )

    const machineContext = computed<Omit<checkbox.Context, "id">>(() => {
      const _inheritedFormControlProps = pick(
        inheritedFormControlProps?.value || {},
        ["isRequired", "isDisabled", "isInvalid", "isReadOnly"]
      )

      const cleanedContextProps = filterUndefined({
        readonly: _inheritedFormControlProps?.isReadOnly?.value,
        required: _inheritedFormControlProps?.isRequired?.value,
        disabled: _inheritedFormControlProps?.isDisabled?.value,
        invalid: _inheritedFormControlProps?.isInvalid?.value,
      })

      const cleanedOwnProps = filterUndefined({
        required: props.isRequired,
        disabled: props.isDisabled,
        invalid: props.isInvalid,
        readonly: props.isReadonly,
        focusable: props.isFocusable,
        "aria-invalid": props["aria-invalid"],
        "aria-label": props["aria-label"],
        "aria-labelledby": props["aria-labelledby"],
        "aria-describedby": props["aria-describedby"],
      })

      return mergeWith({}, cleanedContextProps, cleanedOwnProps, {
        value: props.value,
        defaultChecked: props.defaultChecked,
        indeterminate: props.isIndeterminate,
      })
    })

    const context = computed<checkbox.Context>(() => ({
      id,
      ids: {
        root: rootId.value,
        input: inputId.value,
        control: controlId.value,
        labelId: labelId.value,
      },
      ...machineContext.value,
    }))

    const [state, send] = useMachine(
      // @ts-ignore
      checkbox.machine({ ...context.value })
    )

    const api = computed(() =>
      // @ts-ignore
      checkbox.connect(state.value, send, normalizeProps)
    )

    const iconStyles = computed<SystemStyleObject>(() => ({
      opacity: api.value.isChecked || api.value.isIndeterminate ? 1 : 0,
      transform:
        api.value.isChecked || api.value.isIndeterminate
          ? "scale(1)"
          : "scale(0.95)",
      fontSize: props.iconSize,
      color: props.iconColor,
      ...styles.value.icon,
    }))

    const clonedIcon = computed(() =>
      cloneVNode(h(CheckboxIcon), {
        __css: iconStyles.value,
        isIndeterminate: api.value.isIndeterminate,
        isChecked: api.value.isChecked,
      })
    )

    if (props.defaultChecked && api.value) {
      api.value.setChecked(true)
    }

    if (api.value) {
      if (group.value.value && ownProps.value.value) {
        const isChecked = group.value.value.includes(ownProps.value.value)
        api.value.setChecked(isChecked)
      }
    }

    watch(
      () => api.value.isChecked,
      (value) => {
        emit("update:modelValue", value)
        emit("change", value)
        if (group.value.handleChange && ownProps.value.value) {
          group.value.handleChange(ownProps.value.value, value)
        }
      }
    )

    watch(
      () => props.isIndeterminate,
      (value) => {
        if (api.value && !api.value.isReadOnly) {
          api.value.setIndeterminate(value!)
        }
      }
    )

    watch(
      () => props.modelValue,
      (value) => {
        api.value.setChecked(value)
      }
    )

    const inputProps = computed(() =>
      mergeProps(api.value.inputProps, {
        onChange() {
          if (api.value.isIndeterminate && !api.value.isReadOnly) {
            api.value.setIndeterminate(false)
            api.value.setChecked(true)
          }
        },
      })
    )

    return () => {
      const children = getValidChildren(slots)
      const hasChildren = children.length > 0

      return (
        <chakra.div
          sx={{
            "> div": styles.value.container,
          }}
        >
          {() => (
            <CLabel
              __label="checkbox"
              {...api.value.rootProps}
              __css={styles.value.container}
            >
              {() => (
                <>
                  <input class="chakra-checkbox__input" {...inputProps.value} />
                  <CCheckboxControl
                    __label="checkbox__control"
                    __css={styles.value.control}
                    {...api.value.controlProps}
                  >
                    {() => clonedIcon.value}
                  </CCheckboxControl>
                  {hasChildren && (
                    <chakra.span
                      label="checkbox__label"
                      {...api.value.labelProps}
                      __css={{
                        marginStart: props.spacing,
                        ...styles.value.label,
                      }}
                    >
                      {slots}
                    </chakra.span>
                  )}
                </>
              )}
            </CLabel>
          )}
        </chakra.div>
      )
    }
  },
})
