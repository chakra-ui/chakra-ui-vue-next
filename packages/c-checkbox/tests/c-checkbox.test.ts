// @ts-ignore
import { render } from "@chakra-ui/vue-test-utils"
// @ts-ignore
import { CCheckbox } from "../src"

it("should render properly", () => {
  const { asFragment } = render(CCheckbox)
  expect(asFragment()).toMatchSnapshot()
})
