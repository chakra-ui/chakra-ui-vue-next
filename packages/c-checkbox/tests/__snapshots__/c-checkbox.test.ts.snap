// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly 1`] = `
<DocumentFragment>
  <div
    class="chakra-msv88v"
  >
    <label
      class="chakra-checkbox chakra-1tw369l"
      data-part="root"
      data-scope="checkbox"
      for="input-chakra-checkbox-EMPTY_STRING"
      id="root-chakra-checkbox-EMPTY_STRING"
    >
      <input
        aria-invalid="false"
        aria-labelledby="checkbox:EMPTY_STRING:label"
        class="chakra-checkbox__input"
        data-part="input"
        data-scope="checkbox"
        defaultchecked="false"
        id="input-chakra-checkbox-EMPTY_STRING"
        style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap; word-wrap: normal;"
        type="checkbox"
      />
      <span
        aria-hidden="true"
        class="chakra-checkbox__control chakra-drpdfl"
        data-part="control"
        data-scope="checkbox"
        id="control-chakra-checkbox-EMPTY_STRING"
      >
        <svg
          class="chakra-1mg6af5"
          style="fill: none; stroke-width: 2; stroke: currentColor; stroke-dasharray: 16;"
          viewBox="0 0 12 10"
        >
          <polyline
            points="1.5 6 4.5 9 10.5 1"
          />
        </svg>
      </span>
      <!---->
    </label>
  </div>
</DocumentFragment>
`;
